src/brs/grpc/proto/BrsService.java

```diff
@@ -17,90 +21,295 @@
+        handlerMap.put(GetOrderHandler.class, new GetOrderHandler(assetExchange));
+        handlerMap.put(GetOrdersHandler.class, new GetOrdersHandler(assetExchange));
+        handlerMap.put(GetPeersHandler.class, new GetPeersHandler());
+        handlerMap.put(GetStateHandler.class, new GetStateHandler(timeService, blockchain, generator, blockchainProcessor, propertyService));
+        handlerMap.put(GetSubscriptionHandler.class, new GetSubscriptionHandler(subscriptionService));
+        handlerMap.put(GetSubscriptionsToAccountHandler.class, new GetSubscriptionsToAccountHandler(subscriptionService));
+        handlerMap.put(GetTransactionBytesHandler.class, new GetTransactionBytesHandler(blockchain));
         handlerMap.put(GetTransactionHandler.class, new GetTransactionHandler(blockchain, transactionProcessor));
-        handlerMap.put(GetTransactionBytesHandler.class, new GetTransactionBytesHandler(blockchain, transactionProcessor));
+        handlerMap.put(GetUnconfirmedTransactionsHandler.class, new GetUnconfirmedTransactionsHandler(indirectIncomingService, transactionProcessor));
+        handlerMap.put(ParseTransactionHandler.class, new ParseTransactionHandler());
+        handlerMap.put(SubmitNonceHandler.class, new SubmitNonceHandler(blockchain, accountService, generator));
+        handlerMap.put(SuggestFeeHandler.class, new SuggestFeeHandler(feeSuggestionCalculator));
         this.handlers = Collections.unmodifiableMap(handlerMap);
     }
 
-    private <T extends GrpcApiHandler<?,?>> T getHandler(Class<T> handlerClass) throws HandlerNotFoundException {
-        GrpcApiHandler<?, ?> handler = handlers.get(handlerClass);
-        if (!handlerClass.isInstance(handler)) {
-            throw new HandlerNotFoundException();
+    private <Handler extends GrpcApiHandler<Request, Response>, Request extends Message, Response extends Message> void handleRequest(Class<Handler> handlerClass, Request request, StreamObserver<Response> response) {
+        GrpcApiHandler<? extends Message, ? extends Message> handler = handlers.get(handlerClass);
+        if (handlerClass != null && handlerClass.isInstance(handler)) {
+            Handler handlerInstance = handlerClass.cast(handler);
+            handlerInstance.handleRequest(request, response);
+        } else {
+            response.onError(ProtoBuilder.buildError(new HandlerNotFoundException("Handler not registered: " + handlerClass)));
         }
-        return handlerClass.cast(handler);
     }
 
     @Override
     public void getMiningInfo(Empty request, StreamObserver<BrsApi.MiningInfo> responseObserver) {
-        try {
-            getHandler(GetMiningInfoHandler.class).handleRequest(request, responseObserver);
-        } catch (HandlerNotFoundException e) {
-            responseObserver.onError(ProtoBuilder.buildError(e));
-        }
+        handleRequest(GetMiningInfoHandler.class, request, responseObserver);
```

Comment:
:clap:

Reply:
Took some confusing generics but got it figured out.